UI/importers/studio.cpp

```diff
@@ -0,0 +1,218 @@
+
+		ClearTranslation("wasapi_input_capture",
+				 "coreaudio_input_capture");
+		ClearTranslation("wasapi_output_capture",
+				 "coreaudio_output_capture");
+		ClearTranslation("pulse_input_capture",
+				 "coreaudio_input_capture");
+		ClearTranslation("pulse_output_capture",
+				 "coreaudio_output_capture");
+
+		ClearTranslation("jack_output_capture",
+				 "coreaudio_output_capture");
+		ClearTranslation("alsa_input_capture",
+				 "coreaudio_input_capture");
+
+		ClearTranslation("dshow_input", "av_capture_input");
+		ClearTranslation("v4l2_input", "av_capture_input");
+
+		ClearTranslation("xcomposite_input", "window_capture");
+
+		if (id == "monitor_capture") {
+			if (settings["show_cursor"].is_null() &&
+			    !settings["capture_cursor"].is_null()) {
+				bool cursor =
+					settings["capture_cursor"].bool_value();
+
+				settings["show_cursor"] = cursor;
+			}
+		}
+
+		DirectTranslation("xshm_input", "monitor_capture");
+#elif defined(_WIN32)
+		DirectTranslation("text_ft2_source", "text_gdiplus");
+
+		ClearTranslation("syphon-input", "game_capture");
+
+		ClearTranslation("coreaudio_input_capture",
+				 "wasapi_input_capture");
+		ClearTranslation("coreaudio_output_capture",
+				 "wasapi_output_capture");
+		ClearTranslation("pulse_input_capture", "wasapi_input_capture");
+		ClearTranslation("pulse_output_capture",
+				 "wasapi_output_capture");
+
+		ClearTranslation("jack_output_capture",
+				 "wasapi_output_capture");
+		ClearTranslation("alsa_input_capture", "wasapi_input_capture");
+
+		ClearTranslation("av_capture_input", "dshow_input");
+		ClearTranslation("v4l2_input", "dshow_input");
+
+		ClearTranslation("xcomposite_input", "window_capture");
+
+		if (id == "monitor_capture" || id == "xshm_input") {
+			if (!settings["show_cursor"].is_null()) {
+				bool cursor =
+					settings["show_cursor"].bool_value();
+
+				settings["capture_cursor"] = cursor;
+			}
+
+			source["id"] = "monitor_capture";
+		}
+#else
+		DirectTranslation("text_gdiplus", "text_ft2_source");
+
+		ClearTranslation("coreaudio_input_capture",
+				 "pulse_input_capture");
+		ClearTranslation("coreaudio_output_capture",
+				 "pulse_output_capture");
+		ClearTranslation("wasapi_input_capture", "pulse_input_capture");
+		ClearTranslation("wasapi_output_capture",
+				 "pulse_output_capture");
+
+		ClearTranslation("av_capture_input", "v4l2_input");
+		ClearTranslation("dshow_input", "v4l2_input");
+
+		ClearTranslation("window_capture", "xcomposite_input");
+
+		if (id == "monitor_capture") {
+			source["id"] = "xshm_input";
+
+			if (settings["show_cursor"].is_null() &&
+			    !settings["capture_cursor"].is_null()) {
+				bool cursor =
+					settings["capture_cursor"].bool_value();
+
+				settings["show_cursor"] = cursor;
+			}
+		}
+#endif
+		source["settings"] = settings;
+		sources[i] = source;
+#undef DirectTranslation
+#undef ClearTranslation
+	}
+
+	res = out;
+}
+
+bool StudioImporter::Check(const string &path)
```

Comment:
I feel like you could use Json11 for this function because you're not actually using any stuff that takes in obs_data.  Favor RAII over non-RAII equivalents when possible in C++ code.